src/controllers/controller.line.js

```diff
@@ -201,9 +201,9 @@ module.exports = DatasetController.extend({
 			for (i = 0, ilen = points.length; i < ilen; ++i) {
 				model = points[i]._model;
 				controlPoints = helpers.splineCurve(
-					helpers.previousItem(points, i)._model,
+					points[Math.max(0, i - 1)]._model,
```

Comment:
`i === 0 ? 0 : i - 1` should be faster?

Reply:
Could also do first and Last outside the loop to remove all conditionals from loop.

Reply:
If I remember correctly, we found awhile back that replacing a ternary operator with `Math.min` and `Math.max` in `determineDataLimits` was faster

Reply:
I just updated `element.line.js` because it turns out we didn't even need to do the `Math.min` check there. I'll hold off on moving the other conditionals outside the loop for now because that was the one on the critical path. The others mostly happen when you're using the Bezier options and there's probably larger factors affecting performance in those code paths